QualityControl/public/layout/list/page.js

```diff
@@ -74,77 +74,100 @@ function createHeaderOfFolder(model, folder) {
+function layoutCards(model, layouts, searchBy) {
   return layouts.match({
     NotAsked: () => null,
-    Loading: () => h('', 'Loading...'),
-    Failure: () => h('tr', [h('td', 'Unable to retrieve this list of layouts')]),
+    Loading: () => h('div', 'Loading...'),
+    Failure: () => h('div', [h('div.alert.alert-danger', 'Unable to retrieve this list of layouts')]),
     Success: (list) => {
       if (!list || list.length <= 0) {
-        return h('tr', [h('td', 'No layouts found')]);
+        return h('div', [h('.cardGroupRow', 'No layouts found')]);
       }
-      return list.filter((item) => item.name.match(searchBy))
+      return h('.cardGroupRow', list.filter((item) => item.name.match(searchBy))
         .map((layout) => {
-          const key = `key${layout.name}`;
+          const { description, owner_name, name, id } = layout;
           const { isOfficial } = layout;
           const isMinimumGlobal = model.session.access.some((role) => isUserRoleSufficient(role, UserRole.GLOBAL));
-          return h('tr', { key: key }, [
-            h('td', {
-            }, isOfficial ? h('.primary.f4.text-center', [iconBadge(), ' ']) : ' '),
-            h('td.w-20', [
-              h('.flex-row.items-center', [
-                h('a', {
-                  href: `?page=layoutShow&layoutId=${layout.id}`,
-                  onclick: (e) => model.router.handleLinkEvent(e),
-                }, layout.name),
-              ]),
-            ]),
-            h('td.w-30', layout.owner_name),
-            h('td.w-30', layout.description ?? '-'),
-            h('td', h('.text-right', [
-              isMinimumGlobal && h('button.btn.btn-sm', {
-                onclick: () => model.layout.toggleOfficial(layout.id, !layout.isOfficial),
-              }, layout.isOfficial ? 'Un-official' : 'Official'),
-            ])),
+          const toggleOfficialFunction = (id) => model.layout.toggleOfficial(id);
+
+          return h('.p2.card', [
+            cardHeader(isOfficial, id, name, isMinimumGlobal, toggleOfficialFunction),
+            cardBody(owner_name, description),
           ]);
-        });
+        }));
     },
   });
 }
+
+/**
+ * Generates the card header for a layout with interactive elements including official status toggle.
+ * @param {boolean} isOfficial - Indicates if the layout currently has official status
+ * @param {string} id - Unique identifier for the layout
+ * @param {string} name - Display name of the layout
+ * @param {boolean} isMinimumGlobal - Flag indicating if user has global minimum permissions
+ * @param {Function} toggleOfficialFunction - Callback function to handle official status toggle
+ * @returns {vnode} Virtual DOM node representing the layout card header
+ * @example
+ * // Returns a card header with official styling and toggle button
+ * cardHeader(true, '123', 'My Layout', true, toggleOfficial);
+ */
+function cardHeader(isOfficial, id, name, isMinimumGlobal, toggleOfficialFunction) {
```

Comment:
As we have more than a few parameters, passing an object makes the code more readable

```suggestion
function cardHeader({isOfficial, id, name, isMinimumGlobal, toggleOfficialFunction}) {
```